packages/kit/src/core/sync/write_env.js

```diff
@@ -0,0 +1,117 @@
+import { write_if_changed } from './utils.js';
+import path from 'path';
+import { escape } from './utils.js';
+import fs from 'fs';
+
+const autogen_comment = '// this section is auto-generated';
+
+/**
+ * @param {boolean} pub
+ * @param {Record<string, string>} env
+ * @returns {string}
+ */
+function const_declaration_template(pub, env) {
+	return `${autogen_comment}
+
+${Object.entries(env)
+	.map(
+		([k, v]) => `/**
+ * @type {import('$app/env${pub ? '' : '/private'}').${k}}
+ */
+export const ${k} = "${escape(v)}";
```

Comment:
can this just use `JSON.stringify`?

```suggestion
export const ${k} = ${JSON.stringify(v)};
```

or even

```suggestion
export const ${k} = ${s(v)};
```

if we import it from https://github.com/sveltejs/kit/blob/master/packages/kit/src/utils/misc.js

Reply:
You know, I don't think I even knew you could pass anything but an object/array to `JSON.stringify`. I think that should work fine since it always wraps strings in double quotes.